README.md
# override_text_scale_factor

A Flutter `Widget` that overrides the `textScaleFactor` given by `MediaQuery`.

When you're too lazy to make parts of your UI respond to a non-default `textScaleFactor` 
then `OverrideTextScaleFactor` is the widget for you.

## Usage
The package contains a single widget `OverrideTextScaleFactor`. 
By default this widget will set `textScaleFactor` to 1.0 but the field can be overridden to any value.

### Example
```dart
OverrideTextScaleFactor(
    // Note that any widget can be used as child - not only Text widgets
    child: Container(
        child: Text(
            // This text will always have a textScaleFactor of 1.0
            // regardless of device settings
            'Random text', 
        ),
    ),
);
```
If needed, a specific factor can be used to override the `textScaleFactor`.
### Example
```dart
OverrideTextScaleFactor(
    // Setting the textScaleFactor to a specific value
    textScaleFactor: 1.5,
    child: Container(
        child: Text(
            // This text will always have a textScaleFactor of 1.5
            // regardless of device settings
            'Random text', 
        ),
    ),
);
```

## Example Project
Take a look at this [Example Project](https://github.com/sandstrom99/override_text_scale_factor/blob/master/example/lib/example.dart) Project

## Acknowledgements
This package is close to useless but I've found myself needing an `OverrideTextScaleFactor` widget and you might too.



CHANGELOG.md
## [1.0.2] - 25/03/2021

* `OverrideTextScaleFactor` now takes an optional key in its constructor
* Removed `pedantic` dependency

## [1.0.1] - 02/04/2021

* Fixed CHANGELOG ordering

## [1.0.0] - 02/03/2021

* Migrated to Null-Safety

## [0.0.3] - 08/02/2021

* Updated README.md

## [0.0.2] - 20/01/2021.

* Initial Release






lib/src/widget.dart
import 'package:flutter/material.dart';

/// Flutter widget that overrides [textScaleFactor] passed through [MediaQuery].
class OverrideTextScaleFactor extends StatelessWidget {
  /// Creates a [OverrideTextScaleFactor] widget
  const OverrideTextScaleFactor({
    required this.child,
    this.textScaleFactor = 1.0,
    Key? key,
  }) : super(key: key);

  /// Widget to be used as child
  ///
  /// This cannot be null since there would be no
  /// reason to use [OverrideTextScaleFactor] widget then
  final Widget child;

  /// The [textScaleFactor] used to override the systems' font scale
  ///
  /// By default this is set to 1.0
  final double textScaleFactor;

  @override
  Widget build(BuildContext context) {
    return MediaQuery(
      key: key,
      data: MediaQuery.of(context).copyWith(
        textScaleFactor: textScaleFactor,
      ),
      child: child,
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:override_text_scale_factor/override_text_scale_factor.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Override Text Scale Factor',
      home: Scaffold(
        body: Container(
          width: double.infinity,
          height: double.infinity,
          padding: const EdgeInsets.all(32),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              OverrideTextScaleFactor(
                child: Container(
                  width: double.infinity,
                  color: Colors.black12,
                  padding: EdgeInsets.all(16),
                  child: Text(
                    'This text wont scale to device font scale',
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
              const SizedBox(
                height: 24,
              ),
              OverrideTextScaleFactor(
                textScaleFactor: 1.5,
                child: Container(
                  width: double.infinity,
                  color: Colors.black26,
                  padding: EdgeInsets.all(16),
                  child: Text(
                    'This text has a fixed font scale of 1.5',
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
              const SizedBox(
                height: 24,
              ),
              Container(
                width: double.infinity,
                color: Colors.black38,
                padding: EdgeInsets.all(16),
                child: Text(
                  'This text will scale to device font scale',
                  textAlign: TextAlign.center,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}


example/lib/example.dart
import 'package:flutter/material.dart';
import 'package:override_text_scale_factor/override_text_scale_factor.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Override Text Scale Factor',
      home: Scaffold(
        body: Container(
          width: double.infinity,
          height: double.infinity,
          padding: const EdgeInsets.all(32),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              OverrideTextScaleFactor(
                child: Container(
                  width: double.infinity,
                  color: Colors.black12,
                  padding: const EdgeInsets.all(16),
                  child: const Text(
                    'This text wont scale to device font scale',
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
              const SizedBox(
                height: 24,
              ),
              OverrideTextScaleFactor(
                textScaleFactor: 1.5,
                child: Container(
                  width: double.infinity,
                  color: Colors.black26,
                  padding: const EdgeInsets.all(16),
                  child: const Text(
                    'This text has a fixed font scale of 1.5',
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
              const SizedBox(
                height: 24,
              ),
              Container(
                width: double.infinity,
                color: Colors.black38,
                padding: const EdgeInsets.all(16),
                child: const Text(
                  'This text will scale to device font scale',
                  textAlign: TextAlign.center,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
